repository/src/main/kotlin/com/melih/repository/sources/NetworkSource.kt

```diff
@@ -25,14 +18,14 @@ import javax.inject.Provider
  */
 internal class NetworkSource @Inject constructor(
     private val apiImpl: ApiImpl,
-    private val networkInfoProvider: Provider<NetworkInfo>
+    private val networkInfoProvider: Provider<ConnectivityManager>
 ) : Repository() {
     // region Properties
 
     private val isNetworkConnected: Boolean
         get() {
             val networkInfo = networkInfoProvider.get()
-            return networkInfo != null && networkInfo.isConnected
+            return networkInfo.activeNetworkInfo != null && networkInfo.activeNetworkInfo.isConnected
```

Comment:
Instead of providing `ConnectivityManager` itself, you can provide `activeNetworkInfo` in it, this will ommit obtaining network info in two steps.

Reply:
Ok.